pkg/kubelet/apis/deviceplugin/v1alpha/api.proto

```diff
@@ -20,25 +20,26 @@ option (gogoproto.goproto_unrecognized_all) = false;
 // Registration may fail when device plugin version is not supported by
 // Kubelet or the registered resourceName is already taken by another
 // active device plugin. Device plugin is expected to terminate upon registration failure
-service Registration {
-	rpc Register(RegisterRequest) returns (Empty) {}
-}
+//service Registration {
+//	rpc Register(RegisterRequest) returns (Empty) {}
+//}
 
 message RegisterRequest {
 	// Version of the API the Device Plugin was built against
 	string version = 1;
 	// Name of the unix socket the device plugin is listening on
 	// PATH = path.Join(DevicePluginPath, endpoint)
-	string endpoint = 2;
+	//string endpoint = 2;
 	// Schedulable resource name. As of now it's expected to be a DNS Label
-	string resource_name = 3;
+	string resource_name = 2;
```

Comment:
The kubelet shouldn't be telling which resource_name the device plugin has.
You need a RegisterResponse

Reply:
This is for authentication. kubelet will probe a canonical path and will treat part of socket name as resource name. Device plugin can further authenticate that.